scripts/operations_tests/operations_tests.py

```diff
@@ -140,6 +140,9 @@ def run_test(self, test_case):
         test_case.duration, new_image_stack = self.time_operation(image_stack, test_case.op_func, test_case.params)
         file_name = config_manager.save_dir / (test_case.test_name + ".npz")
 
+        if test_case.params.get('pre_run_step') == 'add_nan':
```

Comment:
Looks like this never actually runs, because the `pre_run_step` attribute is outside the `params` in the json (which makes sense because the params are things to pass to the operation).

 `pre_run_step` will need to be added as an attribute in `TestCase` and set in `compare_mode()`